pallets/subtensor/src/utils/misc.rs

```diff
@@ -96,6 +96,58 @@ impl<T: Config> Pallet<T> {
         Self::deposit_event(Event::OwnerHyperparamRateLimitSet(epochs));
     }
 
+    pub fn cancel_deregistration_priority_schedule_for_owner(owner: &T::AccountId) {
+        let nets: sp_std::vec::Vec<NetUid> = SubnetOwner::<T>::iter()
+            .filter_map(|(netuid, acct)| if acct == *owner { Some(netuid) } else { None })
+            .collect();
+
+        for netuid in nets {
+            if SubnetDeregistrationPrioritySchedule::<T>::take(netuid).is_some() {
+                Self::deposit_event(Event::SubnetDeregistrationPriorityScheduleCleared(netuid));
+            }
+        }
+    }
+
+    pub fn enqueue_subnet_deregistration_priority(netuid: NetUid) -> bool {
+        SubnetDeregistrationPriorityQueue::<T>::mutate(|queue| {
+            if queue.iter().any(|&existing| existing == netuid) {
+                false
+            } else {
+                queue.push(netuid);
+                true
+            }
+        })
+    }
+
+    pub fn remove_subnet_from_deregistration_priority_queue(netuid: NetUid) -> bool {
+        SubnetDeregistrationPriorityQueue::<T>::mutate(|queue| {
+            let original_len = queue.len();
+            queue.retain(|&existing| existing != netuid);
+            original_len != queue.len()
+        })
+    }
+
+    pub fn pop_ready_subnet_deregistration_priority() -> Option<NetUid> {
```

Comment:
```suggestion
    pub fn pop_ready_subnet_from_deregistration_priority_queue() -> Option<NetUid> {
```

I'd say we should keep the introduced misc functions to be more descriptive AND consistent in naming